lib/NVS/src/Namespace.cpp

```diff
@@ -89,11 +96,15 @@ esp_err_t Namespace::get_blob(const char* key, std::ostream& result) {
 }
 
 esp_err_t Namespace::set_blob(const char* key, const void* data, const std::size_t length) {
+    esp_err_t err;
     if (!_isOpened) {
         return ESP_FAIL;
     }
-
-    return nvs_set_blob(_handle, key, data, length);
+    err = nvs_set_blob(_handle, key, data, length);
+    if(err != ESP_OK) {
+        return err;
+    }
+    return nvs_commit(_handle);    
```

Comment:
I think that commit shouldn't be done here, it could have a performance impact when writting various blobs in sequential form like so:

```c++
my_nvs.set_blob(..,);
my_nvs.set_blob(...);
my_nvs.set_blob(...)
```

In that case it's much better to give the API user control of this, so they can handle the situation accordingly.